File: Presentation/Models/HttpModels/Responses/FavoriteResponseDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HttpModels.Responses
{
    public class FavoriteResponseDto
    {
        [Required]
        public Guid Id { get; set; }

        [Required]
        public Guid ProductId { get; set; }

        [Required]
        public Guid CustomerId { get; set; }

        [Required]
        [StringLength(100, ErrorMessage = "Product name cannot exceed 100 characters.")]
        public string ProductName { get; set; }

        [Required(ErrorMessage = "Description is required.")]
        [StringLength(1000, ErrorMessage = "Description cannot exceed 1000 characters.")]
        public string ProductDescription { get; set; }

        [Range(0.0, 5.0, ErrorMessage = "Rating must be between 0.0 and 5.0.")]
        public double ProductRating { get; set; }

        [Range(0.01, 10000.00, ErrorMessage = "Price must be between 0.01 and 10,000.00.")]
        public decimal ProductPrice { get; set; }

        public string ProductImageUrl { get; set; }
    }
}

File: Presentation/cosmeticshop.client/src/apiClient/models/favorite-response-dto.ts
 /**
 * 
 *
 * @export
 * @interface FavoriteResponseDto
 */
 export interface FavoriteResponseDto {

    /**
     * @type {string}
     * @memberof FavoriteResponseDto
     */
    id: string;

    /**
     * @type {string}
     * @memberof FavoriteResponseDto
     */
    productId: string;

    /**
     * @type {string}
     * @memberof FavoriteResponseDto
     */
    customerId: string;

    /**
     * @type {string}
     * @memberof FavoriteResponseDto
     */
    productName: string;

    /**
     * @type {string}
     * @memberof FavoriteResponseDto
     */
    productDescription: string;

    /**
     * @type {number}
     * @memberof FavoriteResponseDto
     */
    productRating: number;

    /**
     * @type {number}
     * @memberof FavoriteResponseDto
     */
    productPrice: number;

    /**
     * @type {string}
     * @memberof FavoriteResponseDto
     */
    productImageUrl: string;
}

